### PR TITLE
S2 coupon detail

### PR DESCRIPTION
**Pull request by federicoboschini**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,8 +10,8 @@ local.properties
...
-app/src/main/res/drawable/
 \.idea/
-app/src/main/res/drawable-xxhdpi/
-app/src/main/res/drawable-xhdpi/
-app/src/main/res/drawable-hdpi/
\ No newline at end of file
+app/src/main/res/drawable/
+app/src/main/res/drawable-xxhdpi/logo2.png
+app/src/main/res/drawable-xhdpi/logo2.png
+app/src/main/res/drawable-hdpi/logo2.png
```
Comment (catt-stefano) [inline .gitignore]: Those files shouldn't be gitignored.

File: README.md [hunk 2]
```diff
@@ -138,3 +138,77 @@ Additionaly, if you wish to replace the header image of the permissions request
...
+
+```java
+  // ...
+  Fragment couponFragment = NearITUIBindings.getInstance(YourActivity.this)
+        .createCouponDetailFragmentBuilder()
+        //  here you can call other methods of the builder
+        .build(coupon);
+```
+
+Even if our pop-up dialog is well designed, you can try to do a better job by overriding resources (see [UI Customization](#ui-customization-1)). As the separator is a PNG file you may want to replace it.
```
Comment (catt-stefano) [inline README.md]: No-one can do a better job than our UX designer 😄
Replace it with "If you need to tweak the way your dialog looks, you can override some resources (see ...)"

File: app/src/main/java/com/nearit/ui_bindings_sample/CouponPlainActivity.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ */
+
+public class CouponPlainActivity extends AppCompatActivity {
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_plain_coupon);
+
+        final Coupon validCoupon = new Coupon();
```
Comment (catt-stefano) [inline app/src/main/java/com/nearit/ui_bindings_sample/CouponPlainActivity.java]: It's better to move coupon creation and init in a method and return it.

EDIT: Just realized you are repeating the process in various activities, it's better to move the creation method in a different object, like a CouponFactory and put a comment before calling its method saying that in a real integration scenario, the coupon object comes from the NearIT library

**Review by catt-stefano — CHANGES_REQUESTED** [review]
Besides the comments on the code, rename the sample coupon placeholder icon res name to 'my_custom_placeholder.png'

File: .gitignore (hunk 1 above)
Comment (federicoboschini) [reply .gitignore]: done

File: README.md (hunk 2 above)
Comment (federicoboschini) [reply README.md]: done

File: app/src/main/java/com/nearit/ui_bindings_sample/CouponPlainActivity.java (hunk 3 above)
Comment (federicoboschini) [reply app/src/main/java/com/nearit/ui_bindings_sample/CouponPlainActivity.java]: done

**Pull request merged**